cli/tests/unit/kv_test.ts

```diff
@@ -1600,6 +1600,24 @@ queueTest("queue retries", async (db) => {
   assertEquals(4, count);
 });
 
+queueTest("queue retries with backoffSchedule", async (db) => {
+  let count = 0;
+  const listener = db.listenQueue((_msg) => {
+    count += 1;
+    throw new TypeError("dequeue error");
+  });
+  try {
+    await db.enqueue("test", { backoffSchedule: [1] });
+    await sleep(2000);
+  } finally {
+    db.close();
+    await listener;
```

Comment:
This is sort of a nitpick/question: would it make more sense to use `Promise.withResolvers()` and resolve after the first message arrive on the queue, then close the db and listener? We could wait for resolution with a timeout. Saying that, because I honestly don't know if the test is flaky or it takes up to 2s to deliver that queue msg on CI.